Update with latest fixes from sustrik/libdill (from Aug 28th til Oct 10th 2017)

ratranqu opened the pull request.

MaxDesiatov commented:
@ratranqu @paulofaria I see that conflicts are quite trivial, any chance this is merged any time soon? Thanks.

sustrik committed:
TERM protocol:wq

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Adjust TERM socket object size

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Terminal handshake removed from PFX protocol

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Add tcp_done

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
ipc_done added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
tls_done added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
ws_done added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
term_storage

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
CRLF, HTTP and WS converted to non-virtual done functions

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
hdone removed

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
hdup replaced by hown

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Revert "NaCl: Fresh random nonce is generated for each message"

This reverts commit 3a333ee0fedf4da43ec4a2384b48c5801ea592a7.

sustrik committed:
Revert "Improved NaCl test"

This reverts commit 165f5bdc2dcc612d25cd596949228c41dfca6bfe.

sustrik committed:
Revert "NaCl protocol ported from dsock"

This reverts commit b9b2aab54d88c5bf109e9a88ff1ccaba62061349.

sustrik committed:
hown manpage added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix man makefile

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Remove deadline parameter from chdone()

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Manpage for chdone() added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Manpage for tcp_done added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Add manpage for ipc_done

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Man pages for tls_done, http_done and ws_done added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Manpage for hdone removed

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
PFX protocol renamed to PREFIX

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
CRLF renamed to POSTFIX

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Rename POSTFIX protocol to SUFFIX

Name clash with the printer protocol.

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Safer allocation of handle numbers

Generate handle numbers that haven't been used recently.

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Keep 8 handles unused

This guarantees at least some handle number rotation
even in the worst case.

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
fix hown function

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Closing frame handling in WS protocol

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
ws: status argument changed to int

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
remove i386-specific storage sizes

I have no way to test they actually work.

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
minor beautification

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
All exported symbols are prefixed by dill_

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Structs get dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Macros in libdill.h have dill_ variant

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
struct dill_iolist

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Docs: Mention macro definitions that modify libdill.h

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Disabling raw names in libdillimpl.h

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Change API for suffix protocol

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
SUFFIX protocol accepts arbitrary suffix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix man pages for SUFFIX protocol

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Documentation of ws protocol updated

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Man pages for term_attach and term_attach_mem added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Documentation for TERM protocol added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Updated basics tutorial

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Remove letfover mention of hdone() function

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Remove one more mention of hdone() function

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
docs: reorder topics in map pages' ToC

sustrik committed:
Bring protocol tutoria up to date.

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
iol-manipulating functions get dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
prefix.c uses dill_ prefixes

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
suffix.c uses dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
udp.c uses dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
term.c uses dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Minor beautification

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
tcp.c uses dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
fd.c uses dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
ipc.c uses dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
bsock.c and msock.c are using dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
FAQ about name clashes added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Missing static specifier added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
now.c uses dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
ipaddr.c uses dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
tls.c uses dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
ipc_type -> dill_ipc_type

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
ipc_listener_type -> dill_ipc_listener_type

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
http.c uses dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
ws.c uses dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
handle.c used dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
libdill.c used dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
chan.c and cr.c use dill_ prefix

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Docs: Note about hown failure behaviour

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Sanitize error handling in prefix and suffix protocol initialization

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
tls initialization sanitized

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
TERM protocol initialization sanitized

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
http: sanitized initialization

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
ws intiialization sanitized

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix attach functions for tls, prefix and suffix.

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix attach functions for tls, ws and http

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Minor beautification

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
One more problem fixed in attach functions

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Detach functions for suffix and prefix sanitized

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Minor beautification

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Initial version of DTLS protocol added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
DTLS: msend can accept multiple iolist items

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
EBUSY error implemented for TCP socket

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
chamek now doesn't return ECANCELED

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Unneeded code removed

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
EBUSY implemented for IPC protocol

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
EBUSY implemented for UDP socket

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Function ipaddr_equal added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Tests for ipaddr_equal

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
UDP: If remote is set packets from different addresses are silently
dropped

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Docs: note about UDP socket dropping incoming packets

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Docs: document EBUSY errors

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
DTLS: test for iolist-based send

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
DTLS: EBUSY implemented

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
dill_iolfrom function added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
DTLS: Allow receiving into multiple iolist elements

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Rightsize the storage structures

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Add missing manpages to the makefile

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
coroutine -> dill_coroutine

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Bump ABI version

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
v2.8 added to the website

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Release process updated

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Don't hold allocated rx buffer for idle connections.

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix memleak in iol test

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix memleaks in choose test

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix memleak in term test

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
FAQ section separated from the reference manual

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

handicraftsman committed:
Fixed CMake config

handicraftsman committed:
Added tcp_wrap_fd(_mem) functions

handicraftsman committed:
Fixed CMake

handicraftsman committed:
Added tcp_wrap_fd(_mem) functions

handicraftsman committed:
Added test for tcp_wrap_fd

sustrik committed:
Nickolay Ilyushin added to AUTHORS

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Renamed: wrap_fd -> fromfd

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Hermetic fromfd test

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
fromfd functions take ownership of the fd

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Check the type of the socket in fromfd functions

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
More checks in tcp_fromfd

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Move fd-checking code to fd.c

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
"fromfd" functions added for TCP listener sockets

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Rearrange tcp.c and ipc.c a little

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Minor beautification

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
fromfd class of functions added to IPC protocol

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix reference counting bug in stack.c

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix buf in hown()

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Eli Riggs added to AUTHORS

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Documentation for fromfd functions added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Account for platforms without SO_ACCEPTCONN

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix bug in fd_check

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Another attempt to fix fd_check

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Add a comment.

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Bump ABI version

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Website updated to version 2.9

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

jfsmig committed:
Add the C++ guard

mjolk committed:
fixed steps 8 and 9 in protocol tutorial

sustrik committed:
Merge pull request #157 from mjolk/tutorialfix

fixed steps 8 and 9 in protocol tutorial

sustrik committed:
Merge pull request #158 from jfsmig/M-cpp-guard

Add the C++ guard

sustrik committed:
Add Jean-Francois Smigielski to AUTHORS file

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Documentation regenerated for v2.10

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Bump ABI version

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Try to fix OSX test failures

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Yet one more attempt to fix the OSX problem

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Use THREAD_FALLBACK to see whether the OSX problem persists

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Revert recent experiments

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Remove some repeated code in ctx.c

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Add comments to ctx.c

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Sanity checking in dill_ctx_term_.

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
On OSX don't use atexit() to destroy context.

Unlike on Linux, on OSX destructor registered by pthread_key_create
seems to be run for the main thread.

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
More sanity checking in ctx.x

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Use thread fallback on OSX

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Use thread fallback on OSX

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Yet one more change to ctx.c on OSX

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Remove obsolete test from Travis

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
'go' test split into several smaller tests

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Larger stack in go5 test

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Run go5 test several times

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Website changed for v2.11

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Bump ABI version

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

jadeblaquiere committed:
don't write 0 bytes via SSL_write (as behavior is undefined in OpenSSL)

sustrik committed:
Merge pull request #162 from jadeblaquiere/no_zero_write_to_SSL

don't write 0 bytes via SSL_write (as behavior is undefined in OpenSSL)

sustrik committed:
Add Joseph deBlaquiere to AUTHORS file

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

jadeblaquiere committed:
fix free() logic in dill_http_hclose(), only free if malloc'd

jadeblaquiere committed:
initial client side attach and tcp connect for socks5

sustrik committed:
Merge pull request #163 from jadeblaquiere/hclose_free_http

fix free() logic in dill_http_hclose(), only free if malloc'd

jadeblaquiere committed:
style refinement

jadeblaquiere committed:
remove socks5_close; add socks5_detach

jadeblaquiere committed:
added connectbyname; connect now takes struct ipaddr as input

jadeblaquiere committed:
simplified interface, no socket layer (simply use bytestream)

sustrik committed:
Update docs for v 2.12

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Bump ABI version

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

jadeblaquiere committed:
avoid issue with memset (optimizer issue with builtin_memset?)

sustrik committed:
Merge pull request #165 from jadeblaquiere/builtin_memset_bug

avoid issue with memset (optimizer issue with builtin_memset?)

jadeblaquiere committed:
Merge branch 'libdill' into socks5

jadeblaquiere committed:
Server Implementation for AUTH, CONNECT

jadeblaquiere committed:
replace _server_ with _proxy_ in names, add documentation

jadeblaquiere committed:
add socks5_proxy_recvcommandbyname which passes the name back to the application (i.e. allowing application specific name lookup rather than DNS)

jadeblaquiere committed:
ensure null termination of hostname

jadeblaquiere committed:
remove diagnostic printfs

jadeblaquiere committed:
adding timeouts to diagnose test failure conditions

jadeblaquiere committed:
Merge branch 'socks5' of https://github.com/jadeblaquiere/libdill into socks5

jadeblaquiere committed:
independant busy states for TCP send/recv

jadeblaquiere committed:
Merge branch 'concurrent_tcp' into socks5-tcp

jadeblaquiere committed:
Merge branch 'concurrent_tcp' into socks5-tcp

jadeblaquiere committed:
basic concurrent tcp proxy

jadeblaquiere committed:
No need to lookup address

sustrik committed:
Merge pull request #166 from jadeblaquiere/concurrent_tcp

independent busy states for TCP send/recv

jadeblaquiere committed:
Merge branch 'libdill' into socks5

jadeblaquiere committed:
Merge branch 'socks5-tcp' into socks5

jadeblaquiere committed:
send TCP_DONE to close connections cleanly

jadeblaquiere committed:
Comments and cleanup

jadeblaquiere committed:
diagnostic message to help diagnose Travis build errors

sustrik committed:
ipaddr_remotes function added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Merge branch 'master' of github.com:sustrik/libdill

sustrik committed:
Happy Eyeballs protocol (RFC 8305)

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Merge pull request #164 from jadeblaquiere/socks5

initial client side attach and tcp connect for socks5

sustrik committed:
struct dill_chclause added

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
happy eyeballs test

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix compiler warnings generated by clang

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix compiler warnings produced by gcc

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Disable happy eyeballs tests

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix build on FreeBSD

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Use gcc-7

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Try using gcc 7

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
gcc7

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
gcc7

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
gcc 7

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Trigger travis build

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
trigger travis build

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
trigger travis build

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
fix .travis.yaml

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix travis build

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
disable socks5 test

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
tweak happy eyeballs test

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
tweak happyeyebalss test

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
tweak happy eyeballs test

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
improve happy eyeballs test

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Temporarily remove SOCKS5 docs.

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix man page generation

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

damag committed:
Fix GCC warning with strncpy

GCC 8.2.1 warns that: ‘strncpy’ specified bound depends on the length
of the source argument. The length of username and password is checked
beforehand anyways, so "strncpy(dst, src, strlen(src))" should be the
same as "strcpy(dst, src)".

sustrik committed:
Merge pull request #169 from damag/master

Fix GCC warning with strncpy

sustrik committed:
Remove DTLS protocol

This was an experimental, undocumented protocol and I have no idea
whether it would be useful or not. Let's revisit if there's an
actual interest in this kind of stuff.

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Add Damien Grassart to AUTHORS file

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Tweak happyeyeballs test

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Add documentation for Happy Eyeballs protocol

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Add documentation for ipaddr_remotes function

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Update docs in preparation for 2.13 release.

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Bump ABI version

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix typo in the documentation

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Less memory allocation in happyeyeballs_connect

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

jadeblaquiere committed:
remove external dependecies (e.g. DNS lookup of libdill.org) from socks5 tests

jadeblaquiere committed:
remove lookup of "localhost", apprently even that fails on some Travis systems

sustrik committed:
Merge pull request #170 from jadeblaquiere/socks5_test

remove external dependecies (lookup of libdill.org) from socks5 test

sustrik committed:
Make the crash when coroutine tries to cancel itself more readable

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Merge branch 'master' of github.com:sustrik/libdill

sustrik committed:
Simplify tls.c

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
More rigorous error handling in TLS protocol

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Separate busy state for sending and receiving in IPC protocol

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Adjust fd_recv() so that it can work without rx buffer

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Minor cleanup in fd.c

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
ipc_sendfd & ipc_recvfd

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Improve sendfd/recvfd test

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

damag committed:
Spelling fix

sustrik committed:
Merge pull request #173 from damag/master

Spelling fix

sustrik committed:
Fix buf in suffix protocol

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Merge branch 'master' of github.com:sustrik/libdill

sustrik committed:
Simplify .gitignore

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>

sustrik committed:
Fix CMake build system

Signed-off-by: Martin Sustrik <sustrik@250bpm.com>
